[Mobile] Tests for base server

erkamkavak opened the pull request:
- tests added for baseService class
- login model corrected based on backend endpoints
- error catching added to baseService class

It would be better to try tests in your local(you should run backend server in the background)

smelihportakal commented:
I tried the test on my local environment and it worked, but I couldn't test it again because a user is created during the registration test. When I attempted a second test, it resulted in an exception in backend indicating that the username already exists.

erkamkavak commented:
> I tried the test on my local environment, but I couldn't test it again because a user is created during the registration test. When I attempted a second test, it resulted in an exception in backend indicating that the username already exists.
> 
> Also i change the username and email in the test but it give me this exception:
> 
> ```
> Expected: <200>
>   Actual: <201>
> 
> package:matcher              expect
> test/service_test.dart 28:5  main.<fn>
> ```

First problem is related to registering with the same username and since there is no "delete user" endpoint in backend, I can't fix it. But I added exception catching to prevent unsuccessful tests. For 201 error, since I don't know the difference between 200 and 201 error codes, I removed it and now only the response message is checked.
